providers/google/internal/contacts/write.go

```diff
@@ -0,0 +1,149 @@
+package contacts
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/urlbuilder"
+	"github.com/amp-labs/connectors/internal/datautils"
+	"github.com/amp-labs/connectors/internal/jsonquery"
+	"github.com/amp-labs/connectors/providers/google/internal/core"
+)
+
+func (a *Adapter) buildWriteRequest(ctx context.Context, params common.WriteParams) (*http.Request, error) {
+	operationName := core.OperationCreate
+	if params.RecordId != "" {
+		operationName = core.OperationUpdate
+	}
+
+	endpoint, err := endpoints.Find(operationName, params.ObjectName, params.RecordId)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := a.getURL(endpoint.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	recordData, err := common.RecordDataToMap(params.RecordData)
+	if err != nil {
+		return nil, err
+	}
+
+	if operationName == core.OperationUpdate && params.ObjectName == objectNameMyConnections {
+		attachMyConnectionsWriteQueryParams(url, recordData)
```

Comment:
same note on this about functions returning values than mutating them (my other note was in the read PR)